lib/auth/token_store.dart

```diff
@@ -9,12 +9,14 @@ abstract class TokenStore {
 
   DateTime? get expiry => _token?._expiry;
 
+  String? get kind => _token?._kind;
+
   bool get hasToken => _token != null;
 
   void setToken(
-      String userId, String idToken, String refreshToken, int expiresIn) {
+      String? userId, String idToken, String refreshToken, int expiresIn, [String? kind]) {
```

Comment:
Yes. Because:
1. there is no User in `signInWithCustomToken`
2. It is used as nullable in your code (`TokenStore` -> `String? get userId => _token?._userId;`), but only here was not null, so it was misleading

Reply:
Good catch, it was probably a leftover from when I transitioned to sound null safety.